src/libopensc/pkcs15-coolkey.c

```diff
@@ -679,7 +679,7 @@ static int sc_pkcs15emu_coolkey_init(sc_pkcs15_card_t *p15card)
 			sc_log(card->ctx, "sc_pkcs15emu_object_add() returned %d", r);
 fail:
 		if (key) { sc_pkcs15_free_pubkey(key); }
-
+		if (r < 0) { free(cert_info.value.value); }
```

Comment:
even if we have old ugly code around, it does not mean the new code needs to be ugly.
```suggestion
		if (r < 0) {
			free(cert_info.value.value);
		}
```